stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -290,6 +262,74 @@ public func _unsafeInheritExecutor_withThrowingTaskGroup<ChildTaskResult, GroupR
+///
+/// Structured concurrency is a way to organize your program, and tasks, in such a way that
+/// tasks do not outlive the scope in which they are created. The guarantee that within
+/// a structured task hierarchy, which can span many levels of child tasks,
+/// no child-task will remain running longer than its parent task, simplifies reasoning about resource usage,
+/// and is a powerful mechanism that you can use to write well-behaved concurrent programs.
+///
+/// Structured Concurrency APIs (including task groups and `async let`), will *always* await the
+/// completion of tasks contained within their scope before returning. Specifically, this means that
+/// even if one were to await a single task result and return it from a `withTaskGroup` function body,
+/// the group will automatically await all the remaining tasks before returning:
+///
+///     func takeFirst(actions: [@Sendable () -> Int]) async -> Int? {
+///         await withTaskGroup { group in
+///             for action in actions {
+///                 group.addTask { action() }
+///             }
+///
+///             return await group.next() // return the first action to complete
+///         } // the group will ALWAYS await the completion of all the actions (!)
+///     }
+///
+/// In the above example, even though we return the first collected integer from all actions added to the task group,
+/// the task group will *always*, automatically, await for the completion of all the resulting tasks.
+///
+/// You may use `group.cancelAll()` to signal cancellation to all the remaining in-progress tasks,
+/// however this will not interrupt their execution automatically, as the child tasks will need to cooperatively
+/// react to the cancellation, and potentially return early (if able to).
+///
+/// In order to create un-structured concurrency tasks, you can use ``Task.init``, ``Task.detached`` or ``Task.immediate``.
+///
+/// Task Group Cancellation
+/// =======================
+///
+/// You can cancel a task group and all of its child tasks
+/// by calling the `cancelAll()` method on the task group,
+/// or by canceling the task in which the group is running.
+///
+/// If you call `addTask(name:priority:operation:)` to create a new task in a canceled group,
+/// that task is immediately canceled after creation.
+/// Alternatively, you can call `addTaskUnlessCancelled(name:priority:operation:)`,
+/// which doesn't create the task if the group has already been canceled.
+/// Choosing between these two functions
+/// lets you control how to react to cancellation within a group:
+/// some child tasks need to run regardless of cancellation,
+/// but other tasks are better not even being created
+/// when you know they can't produce useful results.
+///
+/// In non-throwing task groups the tasks you add to a group with this method are nonthrowing,
+/// those tasks can't respond to cancellation by throwing `CancellationError`.
```

Comment:
```suggestion
/// Because the tasks you add to a nonthrowing task group are nonthrowing,
/// those tasks can't respond to cancellation by throwing `CancellationError`.
```


Removed "this method" because this content has moved out of the API reference for a method.  Also, don't hyphenate "non-throwing"; per [Apple style](https://support.apple.com/guide/applestyleguide), non- and other prefixes aren't usually hyphenated.  There are entries there for most prefixes if you need to check

Reply:
Thanks for the reminder about the Apple style for those!